src/folding/circuits/hypernova/utils.rs

```diff
@@ -0,0 +1,129 @@
+use crate::folding::circuits::utils::VecFpVar;
+use ark_ff::PrimeField;
+use ark_r1cs_std::{
+    alloc::{AllocVar, AllocationMode},
+    fields::{fp::FpVar, FieldVar},
+    ToBitsGadget,
+};
+use ark_relations::r1cs::{Namespace, SynthesisError};
+use std::{borrow::Borrow, marker::PhantomData};
+
+/// `GammaVar` is a wrapper around an `FpVar` for $\gamma$.
+#[derive(Clone)]
+pub struct GammaVar<F: PrimeField> {
```

Comment:
How would you see to move all the circuit related types and methods into `hypernova/circuits.rs`?

Reply:
I don't know why, but I can see the move happening in the commit that you link, but if I go to the last PR changes it appears still under `src/folding/circuits/hypernova/utils.rs`
and also if I go to the latest commit of the branch [dev/hypernova-nimfs-verifier-circuit/mul-gammapow-eq-sigma-gadget](https://github.com/privacy-scaling-explorations/folding-schemes/tree/dev/hypernova-nimfs-verifier-circuit/mul-gammapow-eq-sigma-gadget) :eyes:

Reply:
Oh I see what's happening, the `SumMulsGammaPowsEqSigmaGadget` gadget is duplicated at
https://github.com/privacy-scaling-explorations/folding-schemes/blob/005efd1e1448014ef6e8f74a92a41ced10eff6ba/src/folding/hypernova/circuit.rs#L18
and https://github.com/privacy-scaling-explorations/folding-schemes/blob/005efd1e1448014ef6e8f74a92a41ced10eff6ba/src/folding/circuits/hypernova/utils.rs#L40

Reply:
Ah thanks, I think I rebased incorrectly.. [Done](https://github.com/privacy-scaling-explorations/folding-schemes/pull/53/commits/b939374c62ad2dd4c09f329b59ffcb6820e201ab)